frontend/apps/app/app/lib/vectorstore/syncSchemaVectorStore.ts

```diff
@@ -0,0 +1,162 @@
+import { createClient } from '@liam-hq/db'
+import type { Schema } from '@liam-hq/db-structure'
+import type { Tables } from '@liam-hq/db/supabase/database.types'
+import {
+  createSupabaseVectorStore,
+  isSchemaUpdated,
+} from './supabaseVectorStore'
+
+/**
+ * Schema file path data with projects relation
+ * Based on the schema_file_paths table with added projects field from join query
+ */
+type SchemaFilePathData = Pick<
+  Tables<'schema_file_paths'>,
+  'path' | 'format' | 'organization_id'
+> & {
+  projects?: {
+    organization_id: string
+  }
+}
+
+/**
+ * Creates a Supabase client with service role key
+ * @returns Supabase client
+ */
+function createSupabaseClient() {
+  return createClient(
+    process.env.NEXT_PUBLIC_SUPABASE_URL ?? '',
+    process.env.SUPABASE_SERVICE_ROLE_KEY ?? '',
+  )
+}
```

Comment:
In this PR, we're using `import { createClient } from '@liam-hq/db'` in two specific locations instead of the usual `import { createClient } from 'libs/db/server'`. This is because these particular operations don't require cookie-based authentication.